### PR TITLE
CLI user experience improvements updated docs

### PR DESCRIPTION
**Pull request by oxan**
## Description:
Updated docs for esphome/esphome#1805

**Related issue (if applicable):** N/A

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1805

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: guides/cli.rst [hunk 1]
```diff
@@ -162,44 +162,36 @@ See :ref:`mqtt-using_with_home_assistant`.
...
+The ``esphome mqtt-fingerprint <CONFIG>`` command shows the MQTT SSL fingerprints of the remote used
 for SSL MQTT connections. See :ref:`mqtt-ssl_fingerprints`.
 
 ``version`` Command
 -------------------
 
 .. program:: esphome version
 
-The ``esphome <CONFIG> version`` command shows the current ESPHome version and exits.
+The ``esphome version <CONFIG>`` command shows the current ESPHome version and exits.
```
Comment (glmnet) [inline guides/cli.rst]: Is it really required a config here?

File: guides/cli.rst [hunk 2]
```diff
@@ -162,44 +162,36 @@ See :ref:`mqtt-using_with_home_assistant`.
...
-node configuration file. This is useful if you're not using MQTT discovery.
+The ``esphome clean <CONFIG>`` command cleans all build files and can help with some build issues.
 
 ``dashboard`` Command
 ---------------------
 
 .. program:: esphome dashboard
 
-The ``esphome <CONFIG> dashboard`` command starts the ESPHome dashboard server for using ESPHome
+The ``esphome dashboard <CONFIG>`` command starts the ESPHome dashboard server for using ESPHome
```
Comment (glmnet) [inline guides/cli.rst:196]: Worth to mention that config is a folder in this case

File: guides/cli.rst [hunk 3]
```diff
@@ -241,7 +233,7 @@ The ``esphome <CONFIG> logs`` command validates the configuration and shows all
 
     Manually set the client id.
 
-.. option:: --serial-port SERIAL_PORT
+.. option:: --device SERIAL_PORT
 
-    Manually specify a serial port to use. For example ``/dev/cu.SLAB_USBtoUART``.
+    Manually specify a serial port/IP to use. For example ``/dev/cu.SLAB_USBtoUART``.
```
Comment (glmnet) [inline guides/cli.rst:241]: Also worth mention OTA is an option

File: guides/cli.rst (hunk 1 above)
Comment (oxan) [reply guides/cli.rst]: No.

File: guides/cli.rst (hunk 3 above)
Comment (oxan) [reply guides/cli.rst:241]: I've added a note about OTA's, though not here (since this section is about logs ;))

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**